lib/playground.dart

```diff
@@ -297,6 +294,46 @@ class Playground {
     querySelector("#consoletab").setAttribute('selected','');
   }
 
+
+  _handleAutoCompletion(KeyboardEvent e) {
+    // If we're already in completion bail.
+    if (_isCompletionActive) return;
+
+    if (context.focusedEditor == 'dart') {
+      if (e.keyCode == KeyCode.PERIOD) {
+        editor.completionAutoInvoked = true;
+        editor.execCommand("autocomplete");
+        new Timer(const Duration(milliseconds: 500), _handleHelp);
```

Comment:
`CodeMirror.signal(data, "shown");`

There is this signal, but in show_hints.js. I couldn't find out how to catch it though.

I could catch this signal: `CodeMirror.signal(this, "startCompletion", this);` with `cm.onEvent("startCompletion");`. But this signal is too early. The other signal seems to be the right signal, but I think I maybe have to catch it like, `cm.completion.onEvent("shown")`.

But not sure, I think we need to either change the dart wrapper or even the js code. Maybe it is smarter to send a signal somewhere at completion.dart. Not sure how I could do that.